homeassistant/components/climate/sensibo.py

```diff
@@ -0,0 +1,235 @@
+        return self._ac_states['swing']
+
+    @property
+    def swing_list(self):
+        """List of available swing modes."""
+        return self._current_capabilities['swing']
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def is_aux_heat_on(self):
+        """Return true if AC is on."""
+        return self._ac_states['on']
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._current_capabilities['temperatures'][
+            'C' if self.unit_of_measurement == TEMP_CELSIUS else 'F'][
+                'values'][0]
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._current_capabilities['temperatures'][
+            'C' if self.unit_of_measurement == TEMP_CELSIUS else 'F'][
+                'values'][-1]
+
+    @asyncio.coroutine
+    def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        yield from self._client.async_set_ac_state_property(
+            self._id, 'targetTemperature', int(temperature))
+        yield from self.async_update_ha_state(True)
+
+    @asyncio.coroutine
+    def async_set_fan_mode(self, fan):
+        """Set new target fan mode."""
+        yield from self._client.async_set_ac_state_property(
+            self._id, 'fanLevel', fan)
+        yield from self.async_update_ha_state(True)
+
+    @asyncio.coroutine
+    def async_set_operation_mode(self, operation_mode):
+        """Set new target operation mode."""
+        yield from self._client.async_set_ac_state_property(
+            self._id, 'mode', operation_mode)
+        yield from self.async_update_ha_state(True)
+
+    @asyncio.coroutine
+    def async_set_swing_mode(self, swing_mode):
+        """Set new target swing operation."""
+        yield from self._client.async_set_ac_state_property(
+            self._id, 'swing', swing_mode)
+        yield from self.async_update_ha_state(True)
+
+    @asyncio.coroutine
+    def async_turn_aux_heat_on(self):
+        """Turn Sensibo unit on."""
+        yield from self._client.async_set_ac_state_property(
+            self._id, 'on', True)
+        yield from self.async_update_ha_state(True)
+
+    @asyncio.coroutine
+    def async_turn_aux_heat_off(self):
+        """Turn Sensibo unit on."""
+        yield from self._client.async_set_ac_state_property(
+            self._id, 'on', False)
+        yield from self.async_update_ha_state(True)
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve latest state."""
+        try:
+            data = yield from self._client.async_get_device(
+                self._id, _FETCH_FIELDS)
+            self._do_update(data)
+        except aiohttp.client_exceptions.ClientConnectorError:
```

Comment:
what about all the other errors? Shouldn't you just listen for the base error [`ClientError`](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/client_exceptions.py#L18)